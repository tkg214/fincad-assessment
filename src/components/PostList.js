import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import PostPanel from '../containers/PostPanel';

class PostList extends Component {
  componentDidUpdate() {
    this.scrollToBottom();
  }

  scrollToBottom() {
    const node = ReactDOM.findDOMNode(this['_new']);
    if (node) {
      node.scrollIntoView({ behavior: 'smooth' });
    }
  }

  render() {
    return (
      <div data-test="list">
        {this.props.list.map((post, index) => {
          return (
            <PostPanel
              data-test="post"
              key={index}
              {...post}
              ref={ref => (post.active ? (this['_new'] = ref) : null)}
            />
          );
        })}
      </div>
    );
  }
}

PostList.propTypes = {
  list: PropTypes.array.isRequired
};

export default PostList;
