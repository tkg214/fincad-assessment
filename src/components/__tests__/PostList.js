import React from 'react';
import { shallow } from 'enzyme';

import PostList from '../PostList';

test('the list component renders correctly', () => {
  const wrapper = shallowRenderList();
  expect(wrapper).toMatchSnapshot();
});

test('the list component renders the correct amount of children posts', () => {
  const list = [
    { name: 'test 1', body: 'test' },
    { name: 'test 2', body: 'test' },
    { name: 'test 3', body: 'test' }
  ];
  const wrapper = shallowRenderList({ list });
  expect(wrapper.find(sel('post'))).toHaveLength(list.length);
});

function getListProps(props) {
  return {
    list: [],
    ...props
  };
}

function shallowRenderList(props = {}) {
  const listProps = getListProps(props);
  return shallow(<PostList {...listProps} />);
}

function sel(id) {
  return `[data-test="${id}"]`;
}
