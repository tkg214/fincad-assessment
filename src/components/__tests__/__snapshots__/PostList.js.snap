// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the list component renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    data-test="list"
>
    
</div>,
  "nodes": Array [
    <div
      data-test="list"
>
      
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PostList
        list={Array []}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PostList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "list": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          data-test="list"
>
          
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          data-test="list"
>
          
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PostList
    list={Array []}
/>,
}
`;
